### ISSUE TITLE
Cannot share project if top-level folder contains already a project file

### ISSUE THREAD

**Issue by vogella** [eclipse-egit/egit#63]
### Version

Latest

### Operating System

Linux/Unix

### Eclipse version

2024-12

### Bug description

I cannot share a project into a folder a/b if a contains a .project file.

![image](https://github.com/user-attachments/assets/8e97dcf4-c019-44e5-ad67-6b06cdb857c4)

### Actual behavior

Cannot share via Eclipse, need to move the project outside of Eclipse

### Expected behavior

Possibility to share.

**Comment by Bananeweizen** [comment]
Projects in projects are only available since some time, right? The check might be from the time before this change, and we missed to adapt it, I guess.

**Comment by vogella** [comment]
> Projects in projects are only available since some time, right? The check might be from the time before this change, and we missed to adapt it, I guess.

Yes, only available since 2015 ;-) https://blogs.eclipse.org/post/wayne-beaton/nestedhierarchical-view-projects#:~:text=Hierarchical%20view%20of%20projects%20can,be%20shown%20as%20root%20elements.

**Comment by msohn** [comment]
I tried to reproduce:

- create project `OuterProject`
- add a class `org.example.hello.Hello`
- create a nested project `InnerProject` in the project folder of `OuterProject`
- add a class `org.example.hi.Hi`
- shared the `InnerProject` and created repository in its parent folder
- shared the `OuterProject` and created repository in its parent folder

This is the result I got:

<img width="723" height="706" alt="Image" src="https://github.com/user-attachments/assets/d53a41f2-cee5-40db-ac1a-0b40f59bd8c7" />

Maybe you took different steps and the code path you used has an issue ?

**Comment by vogella** [comment]
It happens if you try to share it.

For example, I just tried to add a new plug-in to eclipse.platform.ui

Create a new plug-in named a in your ws.
Select  it -> Team Share Project... and select the eclipse.platform.ui repo as target (or select the bundles folder as target).

<img width="748" height="477" alt="Image" src="https://github.com/user-attachments/assets/9e06e8b5-7055-4c76-b877-c76311ec6b1d" />

I locally have a patch but need to re-learn how to contribute it to your new Gerrit.... Been a while since I used Gerrit.

**Comment by msohn** [comment]
Ok, I see. 

And you try to move it from the workspace to a subfolder of the existing repo in Eclipse and EGit vetoes that ?
I guess your patch removes the check you pointed out above ?

**Comment by vogella** [comment]
The clone EGit did fails to build with mvn clean verify due to a missing jgit in the parallel folder. Also jgit failed to build with mvn clean verify.
